kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -532,6 +540,60 @@ private static <T extends HasMetadata> boolean isStatusFresh(T resource, Functio
         }
     }
 
+    static <T extends CustomResource<?, ?>> Optional<Kafka> getKafka(Context<T> context, String eventSourceName) {
+        return context.getSecondaryResource(Kafka.class, eventSourceName);
+    }
+
+    public static <T extends CustomResource<?, ?>> ResourceCheckResult<T> checkStrimziKafkaRef(T resource,
+                                                                                               Context<T> context,
+                                                                                               String eventSourceName,
+                                                                                               StrimziKafkaRef strimziKafkaRef,
+                                                                                               String path,
+                                                                                               StatusFactory<T> statusFactory) {
+
+        if (context.getClient().getApiGroup("kafka.strimzi.io") == null) {
+            return new ResourceCheckResult<>(statusFactory.newFalseConditionStatusPatch(resource, ResolvedRefs,
+                    Condition.REASON_REFS_NOT_FOUND,
+                    "strimziKafkaRef present but Kafka resource not found"), List.of());
+        }
+
+        if (isKafka(strimziKafkaRef.getRef())) {
+            Optional<Kafka> kafkaOpt = getKafka(context, eventSourceName);
+            if (kafkaOpt.isEmpty()) {
+                return new ResourceCheckResult<>(statusFactory.newFalseConditionStatusPatch(resource, ResolvedRefs,
+                        Condition.REASON_REFS_NOT_FOUND,
+                        path + ": referenced resource not found"), List.of());
+            }
+            else {
+                String listenerName = strimziKafkaRef.getListenerName();
+                if (!isSupportedListenerType(listenerName)) {
```

Comment:
You are confusing name with type.  **Name is a free text value.**  The Strimzi user is free to call their listeners anything they want.  `blancmange` is just fine!

For this PR, all you need to care about is that the listener has `tls: false`.  If `tls: true` then reject it.  Otherwise it is okay.

https://strimzi.io/docs/operators/latest/configuring.html#specifying_listener_types